### DISCUSSION TITLE
improve aggregation performance

### DISCUSSION THREAD

**Discussion by githorse** [KevinVandy/material-react-table#531] [Ideas]
I've performed an informal grouping-and-aggregation benchmark across several popular grids (MUI, [Material React Table](https://www.material-react-table.com/), [raw Tanstack](https://tanstack.com/table/v8), and AG Grid) in this [CodeSandbox example](https://codesandbox.io/p/sandbox/8t1722?file=%2Fsrc%2FMUI.tsx%3A1%2C1). 

**Rough Loading Time (ms)[^1]**

|Rows|MUI    |MRT      |Tanstack  |AG     | Best Possible|
|--------|---------|-----------|------------|----------|-------------------|
|1k      | 250   |500        |250        |50        |50                  |
|10k    | 1000 | 1000     |500        |250      |50                  |
|50k    | 13000|1600     |1200      |800      |150                |  
|100k  |💣[^2]   |2500      |2500      |1200    |300                |

MRT's performance is not bad, but it's noticeably slower than raw Tanstack, which is noticeably slower than AG Grid (which is noticeably slower than the idealized custom solution). 

I would expect most of this performance to be driven by Tanstack's performance, so I've opened the [same discussion there](https://github.com/TanStack/table/discussions/4860), but MRT does seem to be adding a bit of overhead to the calculation. Obviously it's more polished, so a small premium is inevitable.

Questions:
1) Is there anything I can do as a user of MRT to optimize aggregations?
2) Are there any optimizations MRT can do (apart from Tanstack) to bring the performance of aggregations closer to AG Grid's?

[^1]: Most grids don't have a good way to instrument the exact loading time, so these are basically eyeballed and not very accurate. Obviously they'll be different on different machines. Nevertheless the overall trends are clear enough.
[^2]:Somewhere between 50k and 100k records the MUI grid blows up and kills the whole browser tab.

**Comment by KevinVandy** [comment]
This might be due to MRT enabling all TanStack features and row models. Thanks for the benchmark. Will help me look into this more
